src/families/tron/bridge/js.js

```diff
@@ -363,29 +383,74 @@ const getTransactionStatus = async (a, t): Promise<TransactionStatus> => {
       ? BigNumber(0)
       : await getEstimatedFees(a, t);
 
-  const totalSpent = amount.plus(estimatedFees);
+  const totalSpent = amountSpent.plus(estimatedFees);
 
   if (estimatedFees.gt(0)) {
-    const formattedFees = formatCurrencyUnit(
-      getAccountUnit(account),
-      estimatedFees,
-      {
-        showCode: true,
-        disableRounding: true
-      }
-    );
+    const fees = formatCurrencyUnit(getAccountUnit(account), estimatedFees, {
+      showCode: true,
+      disableRounding: true
+    });
 
-    warnings.fee = new FeeRequired(`Estimated fees: ${formattedFees}`);
+    warnings.fee = new FeeRequired("Estimated fees", { fees });
   }
 
-  if (totalSpent.gt(balance)) {
+  if (["send", "freeze"].includes(mode) && totalSpent.gt(balance)) {
     errors.amount = new NotEnoughBalance();
   }
 
+  if (mode === "unfreeze") {
+    const lowerCaseResource = resource ? resource.toLowerCase() : "bandwidth";
+    const now = new Date();
+    const expirationDate: Date = get(
+      a,
+      `resources.frozen.${lowerCaseResource}.expiredAt`,
+      now
+    );
+
+    if (now.getTime() < expirationDate.getTime()) {
+      errors.resource = new UnfreezeNotExpired(null, {
+        until: expirationDate.toISOString()
+      });
+    }
+  }
+
+  if (mode === "vote") {
+    if (votes.length === 0) {
+      errors.vote = new VoteRequired();
+    } else {
+      const superRepresentatives = await getTronSuperRepresentatives();
+      const isValidVoteCounts = votes.every(v => v.voteCount > 0);
+      const isValidAddresses = votes.every(v =>
+        superRepresentatives.some(s => s.address === v.address)
+      );
+
+      if (!isValidVoteCounts) {
+        errors.voteCount = new InvalidVoteCount();
+      }
+
+      if (!isValidAddresses) {
+        errors.voteAddress = new InvalidAddress();
```

Comment:
in my mind, we need to consider the errors' object keys as "UI fields", i'm not sure if all of these are "fields" or at least conceptually an UI component. it sounds like everything could go into the general `"errors.vote"` as the error object itself tells what the error is about.

in our UI we seem to be heading to having two steps (select voters, then set amounts on them), which maybe would be a local state in UI anyway, but still i'm not sure how we could express and where we'll have that error appearing. probably the UI will have to prevent these to happen, and .vote would be the "general pitfall" we would print somewhere.

another idea is to just "let it throw", the semantic is not yet clearly defined, but a thrown error is accepted and would be a general banner error, and that basically means something general is not OK. like network down.